src/main/java/com/github/manolo8/darkbot/gui/titlebar/BackpageTask.java

```diff
@@ -0,0 +1,264 @@
+    private static final Path VERSION_PATH = BACKPAGE_PATH.resolve(".version");
+
+    private static final String EXECUTABLE_NAME = "dark_backpage";
+    private static final String RELEASE_URL = "https://api.github.com/repos/darkbot-reloaded/DarkBackpage/releases/latest";
+
+    private static final Timer VERSION_CHECK_TIMER = Timer.get(Time.HOUR * 12);
+
+    private final Main main;
+    private final BackpageButton button;
+
+    public BackpageTask(Main main, BackpageButton button) {
+        setDaemon(true);
+        this.main = main;
+        this.button = button;
+    }
+
+    @Override
+    public void run() {
+        if (main.backpage.isInstanceValid()) { // open backpage even if sid is KO
+            try {
+                if (canRun()) {
+                    new ProcessBuilder(BACKPAGE_PATH.resolve(EXECUTABLE_NAME).toAbsolutePath().toString(),
+                            "--sid", main.backpage.getSid(),
+                            "--url", main.backpage.getInstanceURI().toString())
+                            .start().waitFor();
+                }
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        } // inform user that instance/sid is not valid?
+        SwingUtilities.invokeLater(() -> button.setEnabled(true));
+    }
+
+    private boolean canRun() {
+        try {
+            Version currentVersion = readVersionFile();
+
+            // have downloaded backpage and version was checked recently
+            return (currentVersion != null && VERSION_CHECK_TIMER.isActive())
+                    || checkVersion(currentVersion);
+        } catch (IOException e) {
+            e.printStackTrace();
+            VERSION_CHECK_TIMER.disarm(); // IOException happen - reset timer
+
+            String message = e instanceof DirectoryClearException ? ((DirectoryClearException) e).getPopupMessage() : e.toString();
+            Popups.of("Backpage exception", message)
+                    .messageType(JOptionPane.ERROR_MESSAGE)
+                    .showAsync();
+        } finally {
+            button.removeProgressBar();
+        }
+
+        return false;
+    }
+
+    private boolean checkVersion(Version current) throws IOException {
+        JProgressBar progressBar = button.createProgressBar();
+
+        ReleaseInfo releaseInfo = ReleaseInfo.get();
+        if (releaseInfo == null) return false;
+
+        Version remoteVersion = releaseInfo.getVersion();
+        ReleaseInfo.Asset asset = releaseInfo.getValidAsset();
+        if (remoteVersion == null || asset == null) return false;
+
+        VERSION_CHECK_TIMER.activate(); // activate the timer here - version will be successfully checked
+        return current != null && current.compareTo(remoteVersion) >= 0
+                || askUserToDownload(progressBar, current, remoteVersion, asset);
+    }
+
+    //return true if is possible to run a backpage
```

Comment:
If you're looking to document what a return is... may as well do it with javadoc format